app/APKServer.m

```diff
@@ -0,0 +1,141 @@
+//
+//  APKServer.m
+//  iSH
+//
+//  Created by Theodore Dubois on 11/21/20.
+//
+
+#import <CFNetwork/CFNetwork.h>
+#import <sys/socket.h>
+#import <netinet/in.h>
+#import "APKServer.h"
+
+@interface APKServer ()
+
+@property NSFileHandle *server;
+
+@end
+
+@interface APKRequest : NSObject
+
+@property NSFileHandle *client;
+@property CFHTTPMessageRef message;
+- (instancetype)initWithClient:(NSFileHandle *)client;
+
+@end
+
+static NSMutableArray<APKRequest *> *extantRequests;
```

Comment:
Why `static` rather than a property on the server?

Reply:
I would rather it not exist, perhaps instead I can have APKRequest have a strong reference to itself and clear it to free itself

Reply:
A server keeping track of its requests seems like a fairly straightforward design to me. Why do you think creating a reference cycle is a better idea?

Reply:
Refactored